tests/storage/hotplug.go

```diff
@@ -1746,6 +1747,104 @@ var _ = SIGDescribe("Hotplug", func() {
 		})
 	})
 
+	// Some of the functions used here don't behave well in parallel (CreateSCSIDisk).
+	// The device is created directly on the node and the addition and removal
+	// of the scsi_debug kernel module could create flakiness in parallel.
+	Context("[Serial]Hotplug LUN disk", Serial, func() {
+		var (
+			nodeName, address, device string
+			pvc                       *corev1.PersistentVolumeClaim
+			pv                        *corev1.PersistentVolume
+			vm                        *v1.VirtualMachine
+		)
+
+		BeforeEach(func() {
+			nodeName = tests.NodeNameWithHandler()
+			address, device = tests.CreateSCSIDisk(nodeName, []string{})
+			By(fmt.Sprintf("Create PVC with SCSI disk %s", device))
+			pv, pvc, err = tests.CreatePVandPVCwithSCSIDisk(nodeName, device, util.NamespaceTestDefault, "scsi-disks", "scsipv", "scsipvc")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			// Delete the scsi disk
+			tests.RemoveSCSIDisk(nodeName, address)
+			Expect(virtClient.CoreV1().PersistentVolumes().Delete(context.Background(), pv.Name, metav1.DeleteOptions{})).NotTo(HaveOccurred())
+			err := deleteVirtualMachine(vm)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("on an offline VM", func() {
+			By("Creating VirtualMachine")
+			vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(nil)).Create(context.Background(), tests.NewRandomVirtualMachine(libvmi.NewCirros(), false))
+			Expect(err).ToNot(HaveOccurred())
+			By("Adding test volumes")
+			pv2, pvc2, err := tests.CreatePVandPVCwithSCSIDisk(nodeName, device, testsuite.GetTestNamespace(nil), "scsi-disks-test2", "scsipv2", "scsipvc2")
+			Expect(err).NotTo(HaveOccurred(), "Failed to create PV and PVC for scsi disk")
+
+			addVolumeVMWithSource(vm.Name, vm.Namespace, getAddVolumeOptions(testNewVolume1, v1.DiskBusSCSI, &v1.HotplugVolumeSource{
+				PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{PersistentVolumeClaimVolumeSource: corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: pvc.Name,
+				}},
+			}, false, true, ""))
+			addVolumeVMWithSource(vm.Name, vm.Namespace, getAddVolumeOptions(testNewVolume2, v1.DiskBusSCSI, &v1.HotplugVolumeSource{
+				PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{PersistentVolumeClaimVolumeSource: corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: pvc2.Name,
+				}},
+			}, false, true, ""))
+
+			By("Verifying the volumes have been added to the template spec")
+			verifyVolumeAndDiskVMAdded(virtClient, vm, testNewVolume1, testNewVolume2)
+
+			By("Removing new volumes from VM")
+			removeVolumeVM(vm.Name, vm.Namespace, testNewVolume1, false)
+			removeVolumeVM(vm.Name, vm.Namespace, testNewVolume2, false)
+
+			verifyVolumeAndDiskVMRemoved(vm, testNewVolume1, testNewVolume2)
```

Comment:
Can we move this into the `AfterEach`?

Reply:
I think having a different number of volumes between tests overcomplicates this a little bit and would look less nice, prefer to leave it like this.